jchuff.c

```diff
@@ -72,12 +72,30 @@ typedef unsigned long long bit_buf_type;
 typedef size_t bit_buf_type;
 #endif
 
+#if defined WITH_SIMD && !(defined __arm__ || defined __aarch64__)
+typedef unsigned long long simd_bit_buf_type;
+#else
+typedef bit_buf_type simd_bit_buf_type;
+#endif
+
```

Comment:
I don't understand the purpose of this.  Why does `WITH_SIMD` matter for the purposes of determining the C bit buffer size?  Why is a different bit buffer type is used for ARM?